### PR TITLE
fix: re-introduce positional audio with appropriate state handling

### LINKED ISSUE #199: Positional Audio component missing from peer player

### PR DESCRIPTION
**Pull request by kanyesthaker**
### Description of Changes
The collisions PR mistakenly removed the `positionalAudio` component in the peer player, preventing positional audio from working. This reverts that, and also fixes a bug where the audio and video useEffects were not triggering properly due to migrating from builtin React state to Jotai.

### Link to Issue
closes #199 

### Testing Procedure
Testing locally [todo]

### CODE REVIEW

File: src/common/playerStore.ts [hunk 1]
```diff
@@ -14,5 +14,24 @@ export const playerMovementDataMapAtom = atom<Map<string, PlayerMovementData>>(
...
 
+export const playerStreamAtom = (key: string) =>
+  atom((get) => {
+    return get(playerStreamMapAtom).get(key);
+  });
+
+export const audioTracksLengthAtom = (key: string) =>
+  atom((get) => {
+    const audioTracks =
+      get(playerStreamMapAtom).get(key)?.getAudioTracks() || [];
```
Comment (shalins) [inline src/common/playerStore.ts:25]: can we use `??` (nil coalesce) instead `||` here? I think that's more idiomatic

File: src/scene/players/UserPlayer.tsx [hunk 2]
```diff
@@ -257,7 +257,13 @@ const UserPlayer = ({ listener }: UserPlayerProps) => {
     try {
       // navigator.mediaDevices.getUserMedia will check for permissions
       // do not need to explicitly request them
-      let params = { audio: true, video: true, echoCancellation: true };
+      let params = {
```
Comment (shalins) [inline src/scene/players/UserPlayer.tsx:260]:
```suggestion
      let constraints = {
```

File: src/scene/players/UserPlayer.tsx [hunk 3]
```diff
@@ -257,7 +257,13 @@ const UserPlayer = ({ listener }: UserPlayerProps) => {
...
+        audio: true,
+        video: true,
+        echoCancellation: true,
+        autoGainControl: true,
+        noiseSuppression: true,
+      };
       let stream = await navigator.mediaDevices.getUserMedia(params);
```
Comment (shalins) [inline src/scene/players/UserPlayer.tsx:267]:
```suggestion
      let stream = await navigator.mediaDevices.getUserMedia(constraints);
```

File: src/scene/players/PeerPlayer.tsx [hunk 4]
```diff
@@ -152,6 +153,9 @@ const PeerPlayer = ({ listener, peerId }: Props) => {
     <>
       <mesh ref={playerVisualMesh}>
         <boxBufferGeometry args={PlayerConstants.playerSize} />
+        {listener && audioRef && audioRef.current !== undefined && (
```
Comment (shalins) [inline src/scene/players/PeerPlayer.tsx:156]:
```suggestion
        {listener && audioRef && audioRef.current && (
```

File: src/scene/players/UserPlayer.tsx [hunk 5]
```diff
@@ -257,7 +257,13 @@ const UserPlayer = ({ listener }: UserPlayerProps) => {
     try {
       // navigator.mediaDevices.getUserMedia will check for permissions
       // do not need to explicitly request them
-      let params = { audio: true, video: true, echoCancellation: true };
+      let params = {
+        audio: true,
+        video: true,
+        echoCancellation: true,
+        autoGainControl: true,
+        noiseSuppression: true,
```
Comment (shalins) [inline src/scene/players/UserPlayer.tsx:265]: make sure this isn't what causes additional latency in audio

Comment (kanyesthaker) [reply src/scene/players/UserPlayer.tsx:265]: been toggling these off and on, they shouldn't -- in fact most stackoverflow threads recommend setting them to false, but it doesn't help the audio quality -- will set them to false for this PR though

Comment (shalins) [reply src/scene/players/UserPlayer.tsx]: cool, thanks for keeping an eye out!

**Pull request merged**